processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -469,7 +469,8 @@ public boolean isAssignableTo(Type other) {
             // the current target accessor can also be a getter method.
             // The following if block, checks if the target accessor should be overruled by an add method.
             if ( cmStrategy == CollectionMappingStrategyPrism.SETTER_PREFERRED
-                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED ) {
+                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED
+                || cmStrategy == CollectionMappingStrategyPrism.TARGET_IMMUTABLE ) {
```

Comment:
@sjaakd thinking a bit about this. `ADDER_PREFERRED` does not mean `list.add(something)`, it means `entity.add(something)`, right? Theoretically, an adder should also work (if the user implements it correctly). 

@sjaakd I would suggest the following

| Option | Only target set-s Available | Only target add- Available | Both set-s / add- Available | No set-s / add- Available | Existing Target(@TargetType) |
| ------------- | ------------- | ------------- | ------------- |  ------------- | ------------- | 
| `TARGET_IMMUTABLE` | set-s | add | set-s | exception | set-s |

We will only throw exception if there is no adder or setter. WDYT?